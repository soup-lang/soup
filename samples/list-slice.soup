list: ['a', 'b', 'c', 'd', 'e', 'f']
console.log(list[0,3])
console.log(list[1,3])
console.log(list[-1,3])
console.log(list[3,1])
console.log(list[,3])
console.log(list[,-2])
console.log(list[2,])
console.log(list[-2,])

console.log('--- 0 ---')

console.log(list[0,5,2])
console.log(list[1,5,2])
console.log(list[1,5,-2])
console.log(list[5,1,2])
console.log(list[-1,3,-1])

console.log('--- 1 ---')

console.log(list[1,,2])
console.log(list[,,2])
console.log(list[,4,2])
console.log(list[1,,-1])
console.log(list[1,,-2])
console.log(list[,3,-2])
console.log(list[,3,-1])

console.log('--- 2 ---')

console.log(list[,,0])
console.log(list[,])
console.log(list[,,-1])
