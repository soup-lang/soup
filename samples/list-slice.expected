[ 'a', 'b', 'c' ]
[ 'b', 'c' ]
[]
[]
[ 'a', 'b', 'c' ]
[ 'a', 'b', 'c', 'd' ]
[ 'c', 'd', 'e', 'f' ]
[ 'e', 'f' ]
--- 0 ---
[ 'a', 'c', 'e' ]
[ 'b', 'd' ]
[]
[]
[ 'f', 'e' ]
--- 1 ---
[ 'b', 'd', 'f' ]
[ 'a', 'c', 'e' ]
[ 'a', 'c' ]
[ 'b', 'a' ]
[ 'b' ]
[ 'f' ]
[ 'f', 'e' ]
--- 2 ---
[ 'a', 'b', 'c', 'd', 'e', 'f' ]
[ 'a', 'b', 'c', 'd', 'e', 'f' ]
[ 'f', 'e', 'd', 'c', 'b', 'a' ]
